zebra-utils/src/bin/scanning-results-reader/main.rs

```diff
@@ -0,0 +1,128 @@
+//! Displays Zebra's scanning results:
+//!
+//! 1. Opens Zebra's scanning storage and reads the results containing scanning keys and TXIDs.
+//! 2. Fetches the transactions by their TXIDs from Zebra using the `getrawtransaction` RPC.
+//! 3. Decrypts the tx outputs using the corresponding scanning key.
+//! 4. Prints the memos in the outputs.
+
+use std::collections::HashMap;
+
+use jsonrpc::simple_http::SimpleHttpTransport;
+use jsonrpc::Client;
+
+use serde_json::value::RawValue;
+use zcash_client_backend::decrypt_transaction;
+use zcash_client_backend::keys::UnifiedFullViewingKey;
+use zcash_primitives::consensus::{BlockHeight, BranchId};
+use zcash_primitives::transaction::Transaction;
+use zcash_primitives::zip32::AccountId;
+use zebra_scan::scan::sapling_key_to_scan_block_keys;
+use zebra_scan::{storage::Storage, Config};
+
+/// Prints the memos of transactions from Zebra's scanning results storage.
+///
+/// Reads the results storage, iterates through all decrypted memos, and prints the them to standard
+/// output. Filters out some frequent and uninteresting memos typically associated with ZECPages.
+///
+/// Notes:
+///
+/// - `#[allow(clippy::print_stdout)]` is set to allow usage of `println!` for displaying the memos.
+/// - This function expects Zebra's RPC server to be available.
+///
+/// # Panics
+///
+/// When:
+///
+/// - The Sapling key from the storage is not valid.
+/// - There is no diversifiable full viewing key (dfvk) available.
+/// - The RPC response cannot be decoded from a hex string to bytes.
+/// - The transaction fetched via RPC cannot be deserialized from raw bytes.
+#[allow(clippy::print_stdout)]
+pub fn main() {
+    let network = zcash_primitives::consensus::Network::MainNetwork;
+    let storage = Storage::new(&Config::default(), zebra_network(&network), true);
+    let mut prev_memo = "".to_owned();
+
+    for (key, _) in storage.sapling_keys_last_heights().iter() {
+        let dfvk = sapling_key_to_scan_block_keys(key, zebra_network(&network))
+            .expect("Scanning key from the storage should be valid")
+            .0
+            .into_iter()
+            .next()
+            .expect("There should be exactly one dfvk");
+
+        let ufvk_with_acc_id = HashMap::from([(
+            AccountId::from(1),
+            UnifiedFullViewingKey::new(Some(dfvk), None).expect("`dfvk` should be `Some`"),
+        )]);
+
+        for (height, txids) in storage.sapling_results(key) {
+            let height = BlockHeight::from_u32(height.0);
+
+            for txid in txids.iter() {
+                let tx = Transaction::read(
+                    &hex::decode(&get_tx_via_rpc(hex::encode(txid.bytes_in_display_order())))
```

Comment:
Optional: This got added recently. You might also need to `use hex::ToHex`.

```suggestion
                    &hex::decode(&get_tx_via_rpc(txid.encode_hex()))
```

Reply:
Done in [c0c2101](https://github.com/ZcashFoundation/zebra/pull/8104/commits/c0c21016f13f91b63060152cdaabd982c5901354).